### DISCUSSION TITLE
API for in-game highscores, achievements, painless multiplayer, etc.

### DISCUSSION THREAD

**Discussion by Aurel300** [ludumdare/ludumdare#2023] [Ideas]
Considering that the new website is already using an API, it could be interesting to allow game developers to use some of its features for their game. Ideas that come to mind:

 - Multiplayer games without registration. The (rare) multiplayer games on LD are generally local multiplayer. When there is an actual online multiplayer game, registration is often needed, and this requires developers to utilise their own infrastructure, not to mention that it is annoying for the players to have to register. Instead, LDJAM API could act as a trusted authorisation service – people playing LD games are generally participants (or at least members) anyway.
 - Highscores and leaderboards. As an extension of the previous point, games could present leaderboards (unified to some extent) of their games, as opposed to having their own infrastructure (again) or people posting their scores e.g. in the comments.
 - Achievements / trophies. To make games more enjoyable for players, achievements could be given using the API for achieving progress / good scores, etc. This part I am the most unsure about, because in a way trophies could be used to manipulate people's votes – but then again, any game would be free to use them …

And there are probably other uses I can't think of right now. (Cross-game communication?)

While I know in a certain sense every Ludum Dare game, especially the compo entries, should be created from scratch in 48 hours and should be judged as a self-contained entity, allowing an API like this would be an awesome way to leverage the huge community and playerbase that builds up around every LD event. I would say an API like this would qualify for use in compo games the same way monolithic engines like Unity already do (i.e. it is a resource that can be creatively used in many different ways).

It would also make the challenge of making social, multiplayer, or cooperative games less technical in nature. Once again, there are not very many online multiplayer games per Ludum Dare, despite their obvious potential.

Thoughts?

**Comment by mikekasprzak** [comment]
On my short list is Multiplayer Game Matchmaking. To me, one of the most immediate problems in this vein we can help solve is to help multiplayer Ludum Dare games find players.

This past week we got to see how effective this could really be. ShaunJS has had his game running on Twitch, which thanks to the new chat sidebar in the tv widget, means anyone can play the game in 2 clicks right on the website. 

![tribaltwitchy](https://cloud.githubusercontent.com/assets/1570313/25505151/8e233f64-2b6e-11e7-84e5-b3a3f3197086.png)

Now I don't know 100% if the high viewership is thanks to us (always 20+ when I look), but I've had several people compliment that the game in the Twitch widget is pretty cool (even a local friend of mine).

So, I would love to see a section of the sidebar dedicated to games that currently have people waiting for people to join them and play together.

Implementation wise this is less about API calls (a little bit about them) but more about something extra on game pages for multiplayer games. A little matchmaking widget.
